### PR TITLE
Update deployment docs

### PR DESCRIPTION
**Pull request by zacharyDez**
Add description and pointers to eoapi-template, eoapi-cdk, eoapi-k8s. Add information on running services with docker-compose to deployment section. Refactor mkdocs deployment to reference sub-repositories and follow similar formatting for k8s vs. cdk approach.

Followed recommendations from @vincentsarago and @sharkinsspatial [here](https://github.com/developmentseed/eoAPI/pull/114#discussion_r1265878767). Let me know if I missed interpreted anything or if you have other suggestions!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,11 +62,15 @@ Note: services might have incompatible dependencies which you can resolve by usi
 
 ## Deployment
 
-This repository has the demonstration runtimes that are consistently updated with new functionality.
+This repository has current runtimes that are consistently updated with new functionality. 
 
-The default runtimes are available through separate repositories ([eoapi-cdk](https://github.com/developmentseed/eoapi-cdk) and [eoapi-k8s](https://github.com/developmentseed/eoapi-k8s)). 
+The services can be deployed locally via docker with `docker-compose up`.
```
Comment (vincentsarago) [inline README.md:67]: maybe we could also add a note to show how to launch the `official` runtimes

```
docker-compose -f docker-compose.yml -f docker-compose.official.yml up stac-fastapi titiler-pgstac tipg 
```

File: README.md [hunk 2]
```diff
@@ -62,11 +62,15 @@ Note: services might have incompatible dependencies which you can resolve by usi
...
 
-A demonstration application is accessible with the repository [eoapi-template](https://github.com/developmentseed/eoapi-template).
+Two IaC repositories are available:
```
Comment (vincentsarago) [inline README.md:69]:
```suggestion
Two IaC (Infrastructure As Code) repositories are available:
```

File: docs/src/deployment.md [hunk 3]
```diff
@@ -3,13 +3,15 @@ hide:
...
+## Via [eoapi-cdk](https://github.com/developmentseed/eoapi-cdk)
 
-## AWS (Lambda)
 
-An example of Cloud Stack is available for AWS (RDS for the database and Lambda for the APIs)
+[eoapi-cdk](https://github.com/developmentseed/eoapi-cdk) is a set of AWS CDK constructs to deploy eoAPI services.
+
+[eoapi-template](https://github.com/developmentseed/eoapi-template) is an AWS CDK app that shows how to configure the [eoapi-cdk](https://github.com/developmentseed/eoapi-cdk) constructs.
+
+An example of Cloud Stack is available for AWS (RDS for the database and Lambda for the APIs).
```
Comment (vincentsarago) [inline docs/src/deployment.md:14]: I think we should remove the following section and replace with a simple `eoapi-cdk` example

Comment (zacharyDez) [reply docs/src/deployment.md:14]: I can't seem to find the equivalent for the eoAPI-k8s version. @alukach @sharkinsspatial could you put together a short getting started example for the eoapi-cdk repo?

Comment (alukach) [reply docs/src/deployment.md:14]: Sorry, I haven't really been following this project that closely.  Doesn't this repo ship with a `eoapi-cdk`-based example [here](https://github.com/developmentseed/eoAPI/blob/main/infrastructure/aws/cdk/app.py)?

**Comment by zacharyDez** [comment]
@vincentsarago seems like it is going a little stale. Can we keep it as is and review deployment for AWS later on based on feedback?

**Review by vincentsarago — APPROVED** [review]

**Pull request merged**